Shared/Store/BaseDataStore.swift

```diff
@@ -388,13 +388,17 @@ extension BaseDataStore {
     private func unlockInternal() {
         guard let loginsStorage = loginsStorage,
             let loginsKey = loginsKey,
-            let salt = salt else { return }
+            let salt = salt,
+            let loginsDatabasePath = loginsDatabasePath else { return }
 
         do {
             try loginsStorage.ensureUnlockedWithKeyAndSalt(key: loginsKey, salt: salt)
             self.storageStateSubject.onNext(.Unlocked)
         } catch let error as LoginsStoreError {
             pushError(error)
+            // If we can not access database with current salt and key, need to delete local database and migrate to replacement salt
+            // This only deletes the local database file, does not delete the user's sync data
+            _ = handleDatabaseAccessFailure(databasePath: loginsDatabasePath, encryptionKey: loginsKey)
```

Comment:
If we fail in this method, what happens? e.g. if we throw an unexpected runtime exception, or if the app can't continue, perhaps we could fail more loudly (e.g. alert) than a crash?

Reply:
If this fails - maybe I could fire off a reset action from within the handleDatabaseAccessFailure function?  I actually changed this so that handleDatabaseAccessFailure doesn't return, as it didn't need to since the salt gets stored locally in the keychain.